compiler/luci/pass/src/ForwardTransposeOpPass.cpp

```diff
@@ -29,26 +29,52 @@ using namespace luci;
 namespace
 {
 
+// Return true if below conditions are met
+// 1. t->perm() is CircleConst
+// 2. t->perm() is S32
+bool check_perm(const CircleTranspose *t)
+{
+  auto perm = dynamic_cast<CircleConst *>(t->perm());
+  if (not perm)
+    return false;
+
+  switch (perm->dtype())
+  {
+    case loco::DataType::S32:
+      for (uint32_t i = 0; i < perm->size<loco::DataType::S32>(); i++)
+      {
+        auto data = perm->at<loco::DataType::S32>(i);
+        // TODO Support not normalized index
+        if (data < 0 or data >= static_cast<int32_t>(t->rank()))
+          return false;
+      }
+      break;
+    // TODO Support S64 data type
+    default:
+      return false;
+  }
+
+  return true;
+}
+
 // Create new Transpose Op including perm
-// Return nullptr if failed
+// Never return nullptr
 CircleTranspose *create_cloned_transpose(CircleTranspose *transpose)
```

Comment:
I made this function never return nullptr. Why? This function creates a new node, so all necessary checks have to be done before this function is invoked.